src/plugins/data/public/ui/query_string_input/query_bar_top_row.tsx

```diff
@@ -230,10 +232,12 @@ export function QueryBarTopRow(props: Props) {
     }
 
     return (
-      <EuiFlexGroup responsive={false} gutterSize="s">
-        {renderDatePicker()}
-        <EuiFlexItem grow={false}>{button}</EuiFlexItem>
-      </EuiFlexGroup>
+      <NoDataPopover storage={storage} showNoDataPopover={props.indicateNoData}>
```

Comment:
//brainpick

Why does `NoDataPopover` contain the rest of the UI elements? Wouldn't it make more sense, if they are neighbors within a common parent?

Reply:
That's how the `EuiTourStep` component works - it wraps the thing it provides guidance for so the popover with the indicator dot can be rendered in the right place.